### PR TITLE
Post training quantization support in TRTorch

### LINKED ISSUE #35: Support TensorRT Post Training Quantization
We want to be able to use TensorRT's PTQ using a PyTorch dataset to support INT8 execution.

### LINKED ISSUE #41: FP16 Execution produces non accurate results
Right now the same workflow to run FP32 models when used in FP16 gives bad results.

### PR DESCRIPTION
**Pull request by narendasan**
This PR will add PTQ support in TRTorch using the TensorRT INT8 Calibrator. Adds a new application to demonstrate PTQ functionality where the program will quantize a VGG16 network trained on CIFAR10 (training recipe provided) 

## API Changes
- `make_int8_calibrator`: Generates a class that uses a provided dataloader and a TensorRT calibrator algorithm to run quantization
- `extra_info.max_batch_size`: Sets `builder.max_batch_size`
- `extra_info.strict_type` -> `extra_info.strict_types`

## Contract Changes
- User is now responsible for ensuring that the input tensors are the right type for TRT and they are on the GPU

## Bug Fixes
- Discovered a FP16 accuracy issue, added a new test case and will push a solution to this branch

Will close #35, should fix #41

### CODE REVIEW

File: core/conversion/conversion.cpp [hunk 1]
```diff
@@ -235,7 +235,7 @@ bool VerifyConverterSupportForBlock(const torch::jit::Block* b) {
         if (!OpSupported(n)) {
             auto schema = n->maybeSchema();
             TRTORCH_CHECK(schema, "Unable to get schema for Node " << util::node_info(n) \
-                                    << " (conversion.AddLayer)");
+                                    << " (conversion.VerifyCoverterSupportForBloxk");
```
Comment (narendasan) [inline core/conversion/conversion.cpp:238]: Typo
